### DISCUSSION TITLE
Windows: Naming file rustdesk-host=HOST-HERE,key=KEY-HERE.exe is not working properly between 2 machines.

### DISCUSSION THREAD

**Discussion by wazerstar** [rustdesk/rustdesk#2133] [General]
**Describe the bug you encountered:**

So using rustdesk-host=HOST-HERE,key=KEY-HERE.exe will give you the correct info under about details.

But using this as plain default settings and not touching any other settings wont help you connect between 2 machines that are not on same network.

You are forced to go into Network settings and add both ID server & Relay Server on both clients before you can access each other.

**What did you expect to happen instead?**

I would expect when you use the documentation and make portable rustdesk file for windows with integrated host and key settings in naming you would not have to adjust further settings to connect between 2 machines that are not the same network, when you have setup Your own server following the tutorial

**How did you install `RustDesk`?**

Nightly Build rustdesk-1.2.0-x86_64-pc-windows-msvc.exe 

**RustDesk version and environment**

Windows 10 Pro x64 and Windows 11 Pro x64

**Comment by wazerstar** [comment]
Why was this moved to discussion rather than a bug @rustdesk ?

**Reply by pedjas** [reply]
If you run exe without host and key in name it would connect to public server. if you run original exe with host and key in name it would connect to specified host. Clients that are not at the same host cannot connect each other.

If you need to connect to two or more self hosted servers, just create several properly named exe files. Each would connect to its own specified host.

Naturally, this works in so called "portable" version. If you have installed version, that goes to systray, You have to type in host and key to change server.

With version 1.1.9. it was not possible to have two clients running at the same time that need to connect to different hosts. I did not test this nightly build yet to see if that is fixed.

There is discussion to include host info in client ID so that client that wants to connect to it can see what server to connect to. It seems it would be done.

**Reply by rustdesk** [reply]
https://github.com/rustdesk/rustdesk/issues/2118

**Reply by wazerstar** [reply]
> If you run exe without host and key in name it would connect to public server. if you run original exe with host and key in name it would connect to specified host. Clients that are not at the same host cannot connect each other.
> 
> If you need to connect to two or more self hosted servers, just create several properly named exe files. Each would connect to its own specified host.
> 
> Naturally, this works in so called "portable" version. If you have installed version, that goes to systray, You have to type in host and key to change server.
> 
> With version 1.1.9. it was not possible to have two clients running at the same time that need to connect to different hosts. I did not test this nightly build yet to see if that is fixed.
> 
> There is discussion to include host info in client ID so that client that wants to connect to it can see what server to connect to. It seems it would be done.

Not sure how I can be misunderstood here?

I'm using 1 custom server.

with 2 different machines they are using the SAME file with the exact same named settings, they cannot connect to each other until I SPECIFIC add the ID and Relay in each client under Network.

If I however is on same Lan setup with both machines they connect fine to each other without adding ID and Relay, AND still using the custom server.

**Reply by pedjas** [reply]
> with 2 different machines they are using the SAME file with the exact same named settings, they cannot connect to each other until I SPECIFIC add the ID and Relay in each client under Network.
> 
> If I however is on same Lan setup with both machines they connect fine to each other without adding ID and Relay, AND still using the custom server.

I do not understand how clients can connect without using ID.

About connecting two clients on different networks, if your server is behind NAT, then, when you install hbbs service, you have to pay attention to -r parameter. It gets bit complicated when NAT is involved. 

Check explanation here: https://pedja.supurovic.net/setting-up-self-hosted-rustdesk-server-on-windows/
